### ISSUE TITLE
Guide Updates 

### ISSUE THREAD

**Issue by DannyS03** [OpenLiberty/guide-rest-client-angular#63]
- [x] The initial build of the frontend in **`Starting the service`** seems unnecessary. 
- [x] Add a **`project configuration`** section (or subsection). Users should be aware of which resources they are generating as it is not intuitive since the OL server is not responsible for building the frontend. 
- [x] In **`Creating the Angular component`** the **app.component.ts** file is updated twice, despite the section suggesting the creation of the component. Updating the same file twice in the same section can also seem redundant (despite the update being in their own subsections).
  > The guide goes into some detail about the way the data is displayed, which I think can also be reduced as the focus should be the http client. If the data was displayed for UX, then a high-level explanation of that syntax is necessary, but the data is only displayed as plain text.

  > Can also consider displaying the returned data in a more meaningful way. 
- ~`front end` -> `frontend`~

**Comment by realModusOperandi** [comment]
- The initial build I believe was added because liberty:dev does not have generate-resources in its dependency tree so there was a possibility that users would not see the initial Angular starter when they visited its URL. If we could get it set up so that changes to the Angular project triggered an automatic rebuild like ng serve that would be ideal, but it may be too complicated to get working. 

Another option would be to rework the guide to completely separate the Angular app from the backend app and have the frontend served via ng serve. This guide was written to follow the AngularJS guide's pattern as much as possible which involved integrating everything into one project.

- The file is updated twice because it contains two classes which ordinarily would be placed in separate files. I was concerned that doing so here would make things harder to follow as the injected class's definition would not be visible when using it. I'd recommend splitting them into separate files rather than combining everything into one big bang update as the description of what to do will get very long and hard to follow if not broken up.

- The explanation of how to display the data is important IMO as it helps give new users that final bridge from data stored in the backend to displaying it in the browser, keeping the markup simple while also demonstrating a few features that new users could find useful (templating and ngFor). I agree the display is plain and boring but if we go to make it more interested we should take care not to bury those concepts of data insertion under a pile of markup for appearance's sake.

- The spelling of "front end" was suggested by ID in the attached docx at https://github.com/OpenLiberty/guide-rest-client-angular/issues/27, has the guidance on that changed?

**Comment by DannyS03** [comment]
Hi, @realModusOperandi, thank you for the response and insight. 
- I think it would be okay if the **`Creating the Angular component`** section asked the user to **create** the file initially, and then update it in the next section, where the next section becomes a main section. 
- Yes the suggestion is right, sorry, I was interrupting the guide wrong when it asks the user to "build the frontend".

**Comment by realModusOperandi** [comment]
Ah yeah, I see what you mean. I do agree it's redundant to say update the file twice. Perhaps it could be condensed into one file update and if the description of all different pieces could be well-organized it wouldn't be too overwhelming to keep track of. It's been awhile since this guide was written, maybe some newer guides have done it in a better way that this once can follow.

**Comment by DannyS03** [comment]
### Updates
#### Current:
<img width="260" alt="image" src="https://user-images.githubusercontent.com/25497083/84412871-dc473e00-abdd-11ea-97aa-cbf46855ea3e.png">


#### Proposed:
<img width="263" alt="image" src="https://user-images.githubusercontent.com/25497083/84412827-cc2f5e80-abdd-11ea-8f6b-a0dde78c40af.png">

---
- I feel it is better if the user is able to create some of the primary files to improve developer experience the overall guide flow.
  - **`Creating the root Angular module`** rather than just **`Importing the HTTP client`**
as the section is focused on the module rather than the HTTP client itself. The client is later explained/used. 
This also allows for the user to create the root module. 

  - **`Creating the Angular service to fetch data`** allows the user to create  `app.component.ts` (still focusing on angular services), then updating the file (still focusing on angular components) in the **`Defining the component to consume the service`**

- **`Creating the Angular component template`** allows the user to create the component template as the default template already included does not hold much significance.

--- 
- The **`Project Configuration`** section seeks to explain to the user how the front end is executed/built and how it incorporates with the backend (similar motive as the reactjs guide).
- The `finish` directory is configured so that the application is entirely pre-built (which better servers the goal of the finish directory) and the user just needs to start the OL server to see the finished application and does not need generate the node resources.
- After the user builds their application, there is an explicit section, **`Building the front end`** that specifies how to build the front end. This is the only time the user would actually need to build the front end. 

---

Review required, @realModusOperandi @gkwan-ibm.

**Comment by gkwan-ibm** [comment]
- SME review in progress

**Comment by realModusOperandi** [comment]
Changes look good to me.

**Comment by gkwan-ibm** [comment]
- quick peer review (end-2-end, hotspot, typo)
  - Mac (Darren)
  - Linux (Griffin)
  - Win (Tiago)

**Comment by gkwan-ibm** [comment]
- peer review completed
- ID review completed
- next final context review

**Comment by DannyS03** [comment]
- Changes merged to master and verified.
